framework/test/unit/modules/validators/api.spec.ts

```diff
@@ -0,0 +1,426 @@
+			).rejects.toThrow('Invalid timestamp');
+		});
+	});
+
+	describe('getGeneratorAtTimestamp', () => {
+		it(`should return if timestamp is greater than genesis block timestamp`, async () => {
+			await genesisDataSubStore.setWithSchema(
+				emptyKey,
+				{ timestamp: genesisTimestamp },
+				genesisDataSchema,
+			);
+			await generatorListSubStore.setWithSchema(
+				emptyKey,
+				{ addresses: generatorList.map(addr => Buffer.from(addr, 'hex')) },
+				generatorListSchema,
+			);
+			apiContext = new APIContext({ stateStore, eventQueue: new EventQueue() });
+
+			expect(
+				await validatorsModule.api.getGeneratorAtTimestamp(apiContext, genesisTimestamp + 100),
+			).toBeInstanceOf(Buffer);
+		});
+
+		it(`should throw if timestamp is less than genesis block timestamp`, async () => {
+			await genesisDataSubStore.setWithSchema(
+				emptyKey,
+				{ timestamp: genesisTimestamp },
+				genesisDataSchema,
+			);
+			apiContext = new APIContext({ stateStore, eventQueue: new EventQueue() });
+
+			await expect(
+				validatorsModule.api.getGeneratorAtTimestamp(apiContext, genesisTimestamp - 100),
+			).rejects.toThrow('Invalid timestamp');
+		});
+	});
+
+	describe('getGeneratorsBetweenTimestamps', () => {
+		it(`should be able to return if input timestamps are valid`, async () => {
+			await genesisDataSubStore.setWithSchema(
+				emptyKey,
+				{ timestamp: genesisTimestamp },
+				genesisDataSchema,
+			);
+			await generatorListSubStore.setWithSchema(
+				emptyKey,
+				{ addresses: generatorList.map(addr => Buffer.from(addr, 'hex')) },
+				generatorListSchema,
+			);
+			apiContext = new APIContext({ stateStore, eventQueue: new EventQueue() });
+
+			expect(
+				await validatorsModule.api.getGeneratorsBetweenTimestamps(
+					apiContext,
+					genesisTimestamp + 10,
+					genesisTimestamp + 50,
+				),
+			).toBeObject();
+		});
+
+		it(`should throw if input timestamps are invalid`, async () => {
+			apiContext = new APIContext({ stateStore, eventQueue: new EventQueue() });
+
+			await expect(
+				validatorsModule.api.getGeneratorsBetweenTimestamps(
+					apiContext,
+					genesisTimestamp + 10,
+					genesisTimestamp + 1,
+				),
+			).rejects.toThrow('Invalid timestamps');
+		});
+
+		it(`should throw if input timestamp is less than genesis timestamp`, async () => {
+			await genesisDataSubStore.setWithSchema(
+				emptyKey,
+				{ timestamp: genesisTimestamp },
+				genesisDataSchema,
+			);
+			await generatorListSubStore.setWithSchema(
+				emptyKey,
+				{ addresses: generatorList.map(addr => Buffer.from(addr, 'hex')) },
+				generatorListSchema,
+			);
+			apiContext = new APIContext({ stateStore, eventQueue: new EventQueue() });
+
+			await expect(
+				validatorsModule.api.getGeneratorsBetweenTimestamps(
+					apiContext,
+					genesisTimestamp - 100,
+					genesisTimestamp + 1,
+				),
+			).rejects.toThrow('Invalid timestamp');
+		});
+	});
+
+	describe('setGeneratorList', () => {
```

Comment:
I think in general, for `setXXX` functions, maybe it's nice to test if it was stored in the `state store` after execution